addons/event/models/event_ticket.py

```diff
@@ -94,17 +114,36 @@ def _compute_seats(self):
                         GROUP BY event_ticket_id, state
                     """
             self.env['event.registration'].flush_model(['event_id', 'event_ticket_id', 'state', 'active'])
-            self.env.cr.execute(query, (tuple(self.ids),))
+            self.env.cr.execute(query, (tuple(tickets_to_compute.ids),))
             for event_ticket_id, state, num in self.env.cr.fetchall():
                 results.setdefault(event_ticket_id, {})[state_field[state]] = num
 
-        # compute seats_available
-        for ticket in self:
+        # Compute seats for slot tickets and non multi slots tickets
+        for ticket in tickets_to_compute:
             ticket.update(results.get(ticket._origin.id or ticket.id, {}))
-            if ticket.seats_max > 0:
-                ticket.seats_available = ticket.seats_max - (ticket.seats_reserved + ticket.seats_used)
+            slot_seats_max = ticket.event_id.seats_max if ticket.slot_id else 0
+            if ticket.seats_max > 0 or slot_seats_max > 0:
+                seats_max = slot_seats_max if ticket.seats_max == 0 else \
+                            ticket.seats_max if slot_seats_max == 0 else \
+                            min(slot_seats_max, ticket.seats_max)
+                seats_available = seats_max - (ticket.seats_reserved + ticket.seats_used)
+                ticket.seats_available = min(seats_available, ticket.slot_id.seats_available) if ticket.slot_id else seats_available
+            ticket.seats_taken = ticket.seats_reserved + ticket.seats_used
+
+        # Compute seats for parent tickets
+        # Done after the slot tickets computation so that their seats are up to date
+        for ticket in self - tickets_to_compute:
+            ticket.seats_available = sum(ticket.slot_ticket_ids.mapped("seats_available"))
+            ticket.seats_reserved = sum(ticket.slot_ticket_ids.mapped("seats_reserved"))
+            ticket.seats_used = sum(ticket.slot_ticket_ids.mapped("seats_used"))
             ticket.seats_taken = ticket.seats_reserved + ticket.seats_used
 
+    @api.depends("parent_ticket_id", "parent_ticket_id.seats_max")
+    def _compute_seats_max(self):
+        for ticket in self:
+            if ticket.parent_ticket_id:
+                ticket.seats_max = ticket.parent_ticket_id.seats_max
+
```

Comment:
If we change event_type on the event, what happens to the previous tickets (hence, to the previous slots tickets) ?